src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypingStyleDiagnosticAnalyzerBase.State.cs

```diff
@@ -0,0 +1,261 @@
+            /// This typically means that the type name occurs in either left hand or right hand side of an assignment.
+            /// </summary>
+            private bool IsTypeApparentInDeclaration(VariableDeclarationSyntax variableDeclaration, SemanticModel semanticModel, TypeStyle stylePreferences, CancellationToken cancellationToken)
+            {
+                var initializer = variableDeclaration.Variables.Single().Initializer;
+                var initializerExpression = GetInitializerExpression(initializer);
+
+                // default(type)
+                if (initializerExpression.IsKind(SyntaxKind.DefaultExpression))
+                {
+                    return true;
+                }
+
+                // literals, use var if options allow usage here.
+                if (initializerExpression.IsAnyLiteralExpression())
+                {
+                    return stylePreferences.HasFlag(TypeStyle.ImplicitTypeForIntrinsicTypes);
+                }
+
+                // constructor invocations cases:
+                //      = new type();
+                if (initializerExpression.IsKind(SyntaxKind.ObjectCreationExpression) &&
+                    !initializerExpression.IsKind(SyntaxKind.AnonymousObjectCreationExpression))
+                {
+                    return true;
+                }
+
+                // explicit conversion cases: 
+                //      (type)expr, expr is type, expr as type
+                if (initializerExpression.IsKind(SyntaxKind.CastExpression) ||
+                    initializerExpression.IsKind(SyntaxKind.IsExpression) ||
+                    initializerExpression.IsKind(SyntaxKind.AsExpression))
+                {
+                    return true;
+                }
+
+                // other Conversion cases:
+                //      a. conversion with helpers like: int.Parse methods
+                //      b. types that implement IConvertible and then invoking .ToType()
+                //      c. System.Convert.Totype()
+                var declaredTypeSymbol = semanticModel.GetTypeInfo(variableDeclaration.Type, cancellationToken).Type;
+
+                var memberName = GetRightmostInvocationExpression(initializerExpression).GetRightmostName();
+                if (memberName == null)
+                {
+                    return false;
+                }
+
+                var methodSymbol = semanticModel.GetSymbolInfo(memberName, cancellationToken).Symbol as IMethodSymbol;
+                if (methodSymbol == null)
+                {
+                    return false;
+                }
+
+                if (memberName.IsRightSideOfDot())
+                {
+                    var typeName = memberName.GetLeftSideOfDot();
+                    return IsPossibleCreationOrConversionMethod(methodSymbol, declaredTypeSymbol, semanticModel, typeName, cancellationToken);
+                }
+
```

Comment:
What about enums (or in fact any public static members of a type)? Like:

```C#
enum Color { Red, Green, Blue }
...
var color = Color.Red;
```

Just to make it tough, what about fancy patterns like `Color c = Colors.Blue` in the WPF case, or event `Brush b = SystemBrushes.Background`?